Modify rule S1527: Add mention about why we keep adopted future reserved words (#1491)
SonarSource/rspec: rules/S1527/description.adoc

* ``++yield++``

Use of these words as identifiers would produce an error in JavaScript ``++strict++`` mode code.

> This list includes keywords that have been adopted by the latest implementations of ECMAScript. We have kept them in the rule because our analyzer runs on legacy JavaScript codebases as well.

Reviewer: I would try to be more depersonalized

```suggestion
This list includes also the keywords that have been adopted by the latest versions of ECMAScript. They are kept to support legacy JavaScript codebases.
```
